sky/global_user_state.py

```diff
@@ -390,6 +393,32 @@ def set_cluster_metadata(cluster_name: str, metadata: Dict[str, Any]) -> None:
         raise ValueError(f'Cluster {cluster_name} not found.')
 
 
+def get_cluster_storage_mounts_metadata(
+        cluster_name: str) -> Optional[Dict[str, Any]]:
+    rows = _DB.cursor.execute(
+        'SELECT storage_mounts_metadata FROM clusters WHERE name=(?)',
+        (cluster_name,))
+    for (storage_mounts_metadata,) in rows:
+        if storage_mounts_metadata is None:
+            return None
+        return pickle.loads(storage_mounts_metadata)
```

Comment:
Should we directly return `Storage.from_metadata(**pickle.loads(storage_mounts_metadata), sync_on_reconstruction=False)` to hide underlying details from the callers?

Reply:
This method returns a key-value pair of {dst : metadata of `storage` object} rather than solely a metadata of `storage` object. I use `Storage.from_meteadata` to restore all the `storage` objects in the dictionary after this method is called from `cloud_vm_ray_backend.py/_get_cluster_storage_mounts_metadata`